test/simulation-test.coffee

```diff
@@ -66,237 +66,125 @@ describe "Simulation", ->
+    # node on each step.
+    describe "for other scenarios", ->
+      scenarios = [
+        # cascade independent and dependent variables (A->B->C)
+        {A: 50, B: null, C: null, AB: "1 * in", BC: "0.1 * in",
+        results: [
+            [50, 50, 5]
+            [50, 50, 5]
+        ]}
+
+        # cascade independent and dependent variables with negative relationship (A->B->C)
+        {A: 50, B: null, C: null, AB: "0.1 * in", BC: "-1 * in",
+        results: [
+            [50, 5, -5]
+            [50, 5, -5]
+        ]}
+
+        # basic collector (A->[B])
+        {A:5, B:"50+", AB: "1 * in",
+        results: [
+          [5, 50]
+          [5, 55]
+        ]}
+
+        # basic collector with feedback (A<->[B])
+        {A:null, B:"50+", AB: "1 * in", BA: "1 * in",
+        results: [
+          [50, 50]
+          [100, 100]
+          [200, 200]
+        ]}
+
+        # three-node graph (>-) with averaging
+        {A: 10, B: 20, C: null, AC: "1 * in", BC: "1 * in",
+        results: [
+            [10, 20, 15]
+            [10, 20, 15]
+        ]}
+
+        # three-node graph (>-) with non-linear averaging
+        {A: 10, B: 20, C: null, AC: "1 * in", BC: "0.1 * in",
+        results: [
+            [10, 20, 6]
+            [10, 20, 6]
+        ]}
+
+        # three-node graph with collector (>-[C])
+        {A: 10, B: 20, C: "0+", AC: "1 * in", BC: "0.1 * in",
+        results: [
+            [10, 20, 0]
+            [10, 20, 12]
+        ]}
+
+        # three-node graph with collector (>-[C]) and negative relationship
+        {A: 10, B: 1, C: "0+", AC: "1 * in", BC: "-1 * in",
+        results: [
+            [10, 1, 0]
+            [10, 1, 9]
+            [10, 1, 18]
+        ]}
+
+        # Stocks and flow example
+        {A: 50, B: "0+", C: null, D: "0+", E: null
+        AB: "1 * in", BC: "1 * in", CB: "-1 * in", CD: "1 * in", DE: "1 * in", ED: "-1 * in"
+        results: [
+            [50, 0, 0, 0, 0]
+            [50, 50, 50, 0, 0]
+            [50, 50, 50, 50, 50]
+        ]}
+
+        # *** Tests for invalid graphs (should all throw errors) ***
+
+        # two-node graph in a loop with no accumulators (A<->B)
+        {A: null, B: null, AB: "1 * in", BA: "1 * in",
+        results: [false]
+        }
+
+        # three-node graph in a circle with no accumulators (A->B->C->A)
+        {A: null, B: null, C: null, AB: "1 * in", BC: "1 * in", CA: "1 * in",
+        results: [false]
+        }
+
+        # three-node graph with two non-accumulators in a loop ([A]->B<->C)
+        {A: "50+", B: null, C: null, AB: "1 * in", BC: "1 * in", CB: "1 * in",
+        results: [false]
+        }
+      ]
+
+      _.each scenarios, (scenario, i) ->
+        it "should compute scenario #{i} correctly", ->
+          nodes = {}
+          for key, value of scenario
+            if key.length == 1
+              isAccumulator = typeof value is "string"
```

Comment:
Yeah, I ought to be more explicit here. The value of "50+" looked clear to me as a human, to differentiate itself clearly from non-accumulators, but then I figured, since only accumulators get string values, why not just check if it's a string?

I'll add the explicit check for the +, to formalize it.